integrations/apify/src/client.ts

```diff
@@ -0,0 +1,435 @@
+          finishedAt: run.finishedAt?.toISOString() || null,
+          defaultDatasetId: run.defaultDatasetId,
+        },
+      };
+    } catch (error) {
+      this.logger.forBot().error('Error in getRunStatus:', error);
+      
+      if (error instanceof ApifyApiError) {
+        const { message, type, statusCode, clientMethod } = error;
+        this.logger.forBot().error('Apify API Error:', { message, type, statusCode, clientMethod });
+        
+        return {
+          success: false,
+          message: `Apify API Error: ${message}`,
+          data: {
+            error: message,
+            type,
+            statusCode,
+            clientMethod,
+          },
+        };
+      }
+
+      return {
+        success: false,
+        message: error instanceof Error ? error.message : 'Error in getAndSyncRunResults',
+        data: {
+          error: error instanceof Error ? error.message : 'Error in getAndSyncRunResults',
+        },
+      };
+    }
+  }
+
+  async getAndSyncRunResults(runId: string, syncTargetPath?: string, kbId?: string) {
+    const targetPath = syncTargetPath || './crawled-content/';
+    try {
+      const run = await this.client.run(runId).get();
+      
+      if (!run) {
+        return {
+          success: false,
+          message: 'Run not found',
+          data: {
+            runId,
+            error: 'Run not found',
+          },
+        };
+      }
+      
+      if (run.status !== 'SUCCEEDED') {
+        return {
+          success: false,
+          message: `Run is not completed. Current status: ${run.status}`,
+          data: {
+            runId: run.id,
+            status: run.status,
+          },
+        };
+      }
+
+      if (!run.defaultDatasetId) {
+        throw new Error('No dataset ID found for completed run');
+      }
+
+      this.logger.forBot().info(`Getting results from dataset: ${run.defaultDatasetId}`);
+
+      const dataset = this.client.dataset(run.defaultDatasetId);
+      let allItems;
+      try {
+        allItems = await this.fetchAllDatasetItems(dataset);
+        this.logger.forBot().info(`Retrieved ${allItems.length} total items from dataset`);
+      } catch (error) {
+        this.logger.forBot().error('Error fetching dataset items:', error);
+        return {
+          success: false,
+          message: 'Error fetching dataset items',
+          data: {
+            error: error instanceof Error ? error.message : 'Error fetching dataset items',
+          },
+        };
+      }
+
+      let filesCreated = 0;
+      
+      try {
+        // Manual sync
+        if (syncTargetPath) {
+          this.logger.forBot().info(`[GET AND SYNC RUN RESULTS] Starting file sync to path: ${targetPath}`);
+          filesCreated = await this.syncContentToBotpress(allItems, targetPath, kbId || "kb-default");
+          this.logger.forBot().info(`[GET AND SYNC RUN RESULTS] File sync completed. Files created: ${filesCreated}`);
```

Comment:
todo: remove handling of no kbId. While we should have a guard checking for it, it should be way upstream.